08_VirtualFileSystem/02_VirtualFileSystem.md

```diff
@@ -8,6 +8,8 @@ To keep our design simple, the features of our VFS driver will be:
 * No extra features like permissions, uid and gid (although we are going to add those fields, they will not be used).
 * The path length will be limited.
 
+This VFS is built on top of the previously defined resource API (see Resource Management). The VFS creates a file-specific `resource_t` for each open file and returns a handle that is then used with the resource manager's generic `read`, `write`, and `close` functions that will perform the proper _handle-to-resource_ lookup and call our VFS functions.
```

Comment:
I think this paints the wrong picture. The resource api and vfs are separate subsystems that interact but the vfs doesnt depend on the resource api, it can (and should) exist without it. The two would only actually interact when something higher level connects them (like a syscall request).

The resource subsystem makes little sense for in-kernel operations, as you can use raw pointers to things, so why incur the extra cost?